### PR TITLE
FreeBSD compat and fixes

### PR DESCRIPTION
**Pull request by fatih-acar**
Several fixes to allow compilation on FreeBSD:
- cmake module fix
- epoll emulation fix

### CODE REVIEW

File: cmake/modules/FindEPOLL.cmake [hunk 1]
```diff
@@ -9,6 +9,11 @@
 # The hint can be given on the command line too:
 #   cmake -DEPOLL_PATH_HINT=/DATA/ERIC/EPOLL /path/to/source
 
+if (FREEBSD)
```
Comment (dang) [inline cmake/modules/FindEPOLL.cmake:12]: Does something need to be added to detect kevent?  Or is kevent always available?

**Review by dang — COMMENTED** [review]
This looks fine, so far.  Just one comment.

File: cmake/modules/FindEPOLL.cmake (hunk 1 above)
Comment (fatih-acar) [reply cmake/modules/FindEPOLL.cmake:12]: According to man kevent(2):
`The kqueue() and kevent() system calls first appeared in FreeBSD 4.1.`
and AFAIK there is no way to disable kevent at compile-time nor runtime,
so it should always be available.

Comment (dang) [reply cmake/modules/FindEPOLL.cmake:12]: Good enough.

**Review by dang — COMMENTED** [review]
This looks good.  Is it finished?  Or is there more incoming?

**Comment by fatih-acar** [comment]
That's it for me, it should compile and work fine on FreeBSD after that

**Pull request merged**